tests/bud.bats

```diff
@@ -1716,6 +1717,30 @@ _EOF
   run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t ${target} -f Dockerfile3 --build-arg=PGDATA=/pgdata --build-arg=UID=17122 --build-arg=CODE=/copr/coprs_frontend --build-arg=USERNAME=praiskup ${TESTSDIR}/bud/build-arg
   run_buildah inspect -f '{{.FromImageID}}' ${target}
   expect_output "$argsid"
+
+  # If build-arg is specifified via the command line and is different form the previous cached build, it should not use the cached layers.
+  # Note, this containerfile does not have any RUN commands and we verify that the ARG steps are being rebuilt when a change is detected.
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t test-img -f Dockerfile4 ${TESTSDIR}/bud/build-arg
+  run_buildah inspect -f '{{.FromImageID}}' test-img
+  initialid="$output"
+
+  # Build the same containerfile again and verify that the cached layers were used
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t test-img-1 -f Dockerfile4 ${TESTSDIR}/bud/build-arg
+  run_buildah inspect -f '{{.FromImageID}}' test-img-1
+  expect_output "$initialid"
+
+  # Set the build-arg flag and verify that the cached layers are not used
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t test-img-2 --build-arg TEST=foo -f Dockerfile4 ${TESTSDIR}/bud/build-arg
+  run_buildah inspect -f '{{.FromImageID}}' test-img-2
+  argsid="$output"
+  [[ "$argsid" != "$initialid" ]]
+
+  # Set the build-arg via an ENV in the local environment and verify that the cached layers are not used
+  export TEST=bar
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --layers -t test-img-3 --build-arg TEST -f Dockerfile4 ${TESTSDIR}/bud/build-arg
+  run_buildah inspect -f '{{.FromImageID}}' test-img-3
+  argsid="$output"
+  [[ "$argsid" != "$initialid" ]]
```

Comment:
do we have similar tests without the `--layers` option?

Reply:
not sure that we need them, WDYT @umohnani8 ?

Reply:
I don't think we need them. When building without the `--layers` flag we always go through and build every step so should not be an issue in that case.